clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -258,33 +324,48 @@ private TaskHealth inferHealthNetworkBindedContainer(
                   .withTargets(
                       new TargetDescription()
                           .withId(containerInstance.getEc2InstanceId())
-                          .withPort(port)));
+                          .withPort(targetGroupPort.get())));
 
       if (describeTargetHealthResult.getTargetHealthDescriptions().isEmpty()) {
+        log.debug("Target health description is empty");
         evictStaleData(task, loadBalancerService);
         continue;
       }
 
+      log.debug(
+          "Target health description is not empty and has a size of {}",
+          describeTargetHealthResult.getTargetHealthDescriptions().size());
       TargetHealthDescription healthDescription =
           describeTargetHealthResult.getTargetHealthDescriptions().get(0);
 
       TaskHealth taskHealth = makeTaskHealth(task, serviceName, healthDescription);
-      return taskHealth;
+      taskHealthList.add(taskHealth);
     }
 
-    return null;
+    return taskHealthList;
+  }
+
+  private Optional<Integer> getTargetGroupHostPort(List<Container> containers, Integer lbHostPort) {
+    if (containers != null && !containers.isEmpty()) {
+      for (Container container : containers) {
+        for (NetworkBinding networkBinding : container.getNetworkBindings()) {
+          if (networkBinding.getHostPort().intValue() == lbHostPort.intValue()) {
+            log.debug("Load balanced hostPort: {} found for container.", lbHostPort);
+            return Optional.of(lbHostPort);
+          }
+        }
+      }
+    }
+
+    return Optional.empty();
   }
```

Comment:
If this (and getTargetContainerPort) just return the port input value if found and empty if not, couldn't these methods just return booleans?  The caller already has the value.

Reply:
That's fair, initially I wanted getTargetContainerPort(getTargetHostPort) to have the logic to find the container port, but currently it's pretty straight forward. Updated